src/libopenrave/kinbodygeometry.cpp

```diff
@@ -234,13 +234,15 @@ bool KinBody::GeometryInfo::InitCollisionMesh(float fTessellation)
     case GT_Cylinder: {
         // cylinder is on z axis
         dReal rad = GetCylinderRadius(), len = GetCylinderHeight()*0.5f;
-        int numverts = (int)(fTessellation*48.0f) + 3;
+        // int numverts = (int)(fTessellation*48.0f) + 3;
+        int numverts = max((int)(fTessellation*GetCylinderNumCircleApproximate()), 3); // should not be less than 3 to maintain backwards compatibility  (numverts = (int)(fTessellation*48.0f) + 3;)
```

Comment:
@kanunikov-denys not sure if we need "tessellation" factor anymore since GeometryInfo is holding the discretized vertex count anymore... I would put this into its own new `int` parameter called `numVerticesWhenDiscretizing`. That way we can use it for spheres and other interesting primitives too. Perhaps we don't even need `fTessellation` in `InitCollisionMesh` anymore~